src/taskManager/UserInterface.java

```diff
@@ -0,0 +1,114 @@
+package taskManager;
+
+import java.util.Scanner;
+
+public class UserInterface {
+    Scanner sc = new Scanner(System.in);
+    TaskList taskList = new TaskList();
+
+    public void startUI(){
+        while(true){
+            printCommandOptions();
+        }
+    }
+
+    public void printCommandOptions() {
+        System.out.println("Select command you would like to implement from following:");
+        System.out.println("1 - add Task");
+        System.out.println("2 - edit Task");
+        System.out.println("3 - remove Task");
+        System.out.println("4 - print Tasks List");
+        System.out.println("5 - change task status");
+        System.out.println("exit - exit the application");
+        handleInput();
+    }
+
+    public void handleInput() {
+        String command = sc.nextLine();
+        switch (command) {
+            case "1":
+                addTask();
+                break;
+            case "2":
+                editTask();
+                break;
+            case "3":
+                removeTask();
+                break;
+            case "4":
+                printTasksList();
+                break;
+            case "5":
+                changeStatus();
+                break;
+            case "exit":
+                exit();
+                break;
+            default:
+                System.out.println("Invalid command");
+
+        }
+    }
+
+    public void addTask() {
+        boolean isEmptyInput = true;
+        String[] taskFields = {"title", "description", "priority", "status"};
+        String[] taskArr =  new String[4];
+        int i = 0;
+        for (String field: taskFields) {
+            String input = "";
+            while (isEmptyInput) {
+                System.out.println(String.format("Enter task %s: ", field));
+                input = sc.nextLine();
+                isEmptyInput = input.equals("");
+            }
+            isEmptyInput = true;
+            taskArr[i] = input;
+            i++;
+        }
+
+        Task task = new Task(taskArr[0], taskArr[1], taskArr[2], taskArr[3]);
```

Comment:
```suggestion
        List<String> taskFieldValues =  new ArrayList<String>(TASK_FIELDS.length);
        for (String field: TASK_FIELDS) {
            while (true) {
                System.out.println(String.format("Enter task %s: ", field));
                var input = sc.nextLine();
                if (!input.equals("")) {
                    taskFieldValues.add(input);
                    break;
                }
            }
        }

        Task task = new Task(taskFieldValues.get(0), taskFieldValues.get(1), taskFieldValues.get(2), taskFieldValues.get(3));
```

Reply:
Thanks, I had a feeling that there should be a more elegant solution. :)